crates/bevy_animation/src/lib.rs

```diff
@@ -123,23 +123,50 @@ impl AnimationClip {
     }
 }
 
+/// Repetition behavior of an animation.
+#[derive(Reflect, Copy, Clone)]
+pub enum RepeatAnimation {
+    /// The animation will never finish.
+    Forever,
+    /// The animation will finish after running once.
+    Never,
+    /// The animation will finish after running "n" times.
+    Count(u32),
+}
+
 #[derive(Reflect)]
 struct PlayingAnimation {
-    repeat: bool,
+    repeat: RepeatAnimation,
     speed: f32,
     elapsed: f32,
-    animation_clip: Handle<AnimationClip>,
+    animation_clip: Option<Handle<AnimationClip>>,
     path_cache: Vec<Vec<Option<Entity>>>,
+    /// Number of times the animation has completed.
+    /// If the animation is playing in reverse, this increments when the animation passes the start.
+    completions: u32,
```

Comment:
Could you add a public accessor to this?

Reply:
Updated